src/EnergyPlus/PlantLoadProfile.cc

```diff
@@ -260,6 +254,71 @@ void PlantProfileData::InitPlantProfile(EnergyPlusData &state)
+            inletTemp = state.dataSize->PlantSizData(plntSizIndex).ExitTemp;
+        }
+        auto &plntComps = state.dataPlnt->PlantLoop(this->plantLoc.loopNum).plantCoilObjectNames;
+        auto &cmpType = state.dataPlnt->PlantLoop(this->plantLoc.loopNum).plantCoilObjectTypes;
+        int arrayIndex = -1;
+        for (size_t i = 0; i < plntComps.size(); ++i) {
+            if (plntComps[i] == this->Name && cmpType[i] == this->Type) {
+                arrayIndex = i;
+                break;
+            }
+        }
+        if (arrayIndex == -1) {
+            state.dataPlnt->PlantLoop(this->plantLoc.loopNum).plantCoilObjectNames.emplace_back(this->Name);
+            state.dataPlnt->PlantLoop(this->plantLoc.loopNum).plantCoilObjectTypes.emplace_back(this->Type);
+            std::vector<Real64> tmpFlowData;
+            tmpFlowData.resize(size_t(24 * state.dataGlobal->TimeStepsInHour + 1));
+            tmpFlowData[0] = -1; // comp index
+            if (this->FluidType == PlantLoopFluidType::Water) {
+                FluidDensityInit = this->plantLoc.loop->glycol->getDensity(state, inletTemp, RoutineName);
+            } else { //(this->FluidType == PlantLoopFluidType::Steam)
+                FluidDensityInit = this->plantLoc.loop->steam->getSatDensity(state, inletTemp, 1.0, RoutineName);
+            }
+            Real64 Cp;
+            if (this->FluidType == PlantLoopFluidType::Water) {
+                Cp = this->plantLoc.loop->glycol->getSpecificHeat(state, inletTemp, RoutineName);
+            } else if (this->FluidType == PlantLoopFluidType::Steam) {
+                Real64 EnthSteamInDry = this->plantLoc.loop->steam->getSatEnthalpy(state, inletTemp, 1.0, RoutineName);
+                Real64 EnthSteamOutWet = this->plantLoc.loop->steam->getSatEnthalpy(state, inletTemp, 0.0, RoutineName);
+                Real64 LatentHeatSteam = EnthSteamInDry - EnthSteamOutWet;
+                Real64 SatTemp = this->plantLoc.loop->steam->getSatTemperature(state, DataEnvironment::StdPressureSeaLevel, RoutineName);
+                Cp = this->plantLoc.loop->glycol->getSpecificHeat(state, SatTemp, RoutineName);
+
+                // Steam Mass Flow Rate Required
+                this->MassFlowRate = this->Power / (LatentHeatSteam + this->DegOfSubcooling * Cp);
+                PlantUtilities::SetComponentFlowRate(state, this->MassFlowRate, this->InletNode, this->OutletNode, this->plantLoc);
+                state.dataLoopNodes->Node(this->OutletNode).Quality = 0.0;
+            }
+            for (size_t i = 1; i <= thisLoadSched.size(); ++i) {
+                if (plntDeltaT > 0) {
+                    tmpFlowData[i] =
+                        thisLoadSched[i - 1] / (FluidDensityInit * Cp * plntDeltaT); // back calculate volume flow based on Q and sizing data
+                } else {
+                    tmpFlowData[i] = thisFlowSched[i - 1] * this->PeakVolFlowRate; // use flow schedule if not a sizing run
+                }
```

Comment:
LoadProfile:Plant also places the load in the comp flow data array, as the Q capacity / (rho * Cp * delta T) to align with the Sizing:Plant object (i.e., the calculated flow is the actual "load" on the loop given Sizing:Plant inputs).